sound/soc/sof/loader.c

```diff
@@ -210,30 +211,51 @@ static int load_modules(struct snd_sof_dev *sdev, const struct firmware *fw)
 int snd_sof_load_firmware_memcpy(struct snd_sof_dev *sdev,
 				 const struct firmware *fw)
 {
+	struct snd_sof_pdata *plat_data = dev_get_platdata(sdev->dev);
 	int ret;
 
+	/* set code loading condition to true */
+	sdev->code_loading = 1;
+
+	ret = request_firmware(&plat_data->fw,
```

Comment:
could we try to optimize the topology loading as well, by requesting the topology file here in no_wait mode and so that we wait less when we deal with topology later?

Reply:
how about doing this change later? change won't take big effort, but I don't want to submit it without test on as most as possible platforms, I will do this after HDMI.

Reply:
ok.